docs/tutorial/dynamic_loading/index.rst

```diff
@@ -0,0 +1,50 @@
+.. _sec-dynamic-loading:
+
+======================================
+Dynamically loading the shared library
+======================================
+
+In this section, you will find an example of how to call the `xnvme_enumerate` function when dynamically loading the **xNVMe** shared library. The example serves to illustrate how one can go about doing this and some common pitfalls to watch out for. The example will first be presented in C and then in python.
+
+To get an overview of what the **xNVMe** shared library contains refer to :ref:`sec-c-api`
+
+Using the shared library in C
+-----------------------------
+Here `dlopen` is used to load the library and pointers to the relevant functions are returned by `dlsym`.
+
+Pitfalls:
+
+Don't dismiss the error handling
+  When `dlsym` can't find a symbol in the library, null is returned and the program will continue.
+  Because of this, a missing symbol can be overlooked and cause the program to behave in unexpected ways or crash.
+  This can lead to a lot of time wasted debugging. All of this can be avoided by immediately checking whether the symbol exists before moving on. 
+
+`dlsym` will not find enumerators
+  An enumerator are an *integer constant expression* and has no reference.
+  Therefore, `dlsym` will not find it. Instead, the enumerators can be used as long the relevant headers are included.  
+
+.. literalinclude:: enumerate_example.c
+   :language: c
+
```

Comment:
It would be great with the inclusion of an example of how to compile and run it, similar to:

- ``docs/getting_started/100_compile.cmd``
- ``docs/getting_started/100_compile.out``
- ``docs/getting_started/110_run.cmd``
- ``docs/getting_started/110_run.out``

When creating a ``.cmd`` file in the docs, then the command within is executed and the output produced stored in ``.out``, during doc-generation, it thus also serve as a way to test that the examples provided in the documentation are correct and "up-to-date".

The first ``.cmd`` could look something like:
```
gcc ./enumerate_example.c -o enumerate_example
```
The second on how to run it:
```
./enumerate_example
```
And then also do a ``literal`` include of the ``.out`` file. Then the reader get's to see the code, how to compile it and expected output, plus it will be "verified" that it works.

Reply:
fixed in commit de4d90d